File: backend/src/recetario/domain/entities/meal.py
"""Meal-calendar domain entities — pure, framework-free.

A `MealEvent` schedules one recipe into a meal slot on a given date. A "week" is
simply the set of events whose date falls in a range — no separate week entity
(see DESIGN.md §2).
"""

from __future__ import annotations

from dataclasses import dataclass
from datetime import date as Date
from datetime import datetime
from decimal import Decimal
from enum import Enum


class MealType(str, Enum):
    BREAKFAST = "breakfast"
    LUNCH = "lunch"
    DINNER = "dinner"
    SNACK = "snack"


@dataclass
class MealEvent:
    """A scheduled recipe in a meal slot.

    `servings_planned` is how many servings of the recipe are planned for this
    slot; macro aggregation scales the recipe's per-serving profile by it.
    `recipe_title` is a read-time convenience populated by the repository for the
    calendar view; it is not persisted on this row.
    """

    date: Date
    meal_type: MealType
    recipe_id: int
    servings_planned: Decimal = Decimal(1)
    notes: str | None = None
    id: int | None = None
    recipe_title: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: core/src/entities/meal.ts
/**
 * Meal-calendar entities. Port of `domain/entities/meal.py`.
 *
 * A `MealEvent` schedules one recipe into a meal slot on a given date. A "week"
 * is simply the events whose date falls in a range — no separate week entity.
 * Dates are ISO `YYYY-MM-DD` strings (sortable, JSON-friendly, valid Map keys).
 */
import type { Decimal } from "decimal.js";

export enum MealType {
  BREAKFAST = "breakfast",
  LUNCH = "lunch",
  DINNER = "dinner",
  SNACK = "snack",
}

export interface MealEvent {
  date: string;
  mealType: MealType;
  recipeId: number;
  servingsPlanned?: Decimal;
  notes?: string | null;
  id?: number | null;
  /** Read-time convenience populated by the repository for the calendar view. */
  recipeTitle?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}
